src/main/groovy/com/inxmail/clicktracker/entity/Url.groovy

```diff
@@ -2,13 +2,28 @@ package com.inxmail.clicktracker.entity
 
 import com.fasterxml.jackson.annotation.JsonFormat
 
+import javax.persistence.Column
+import javax.persistence.Entity
+import javax.persistence.GeneratedValue
+import javax.persistence.GenerationType
+import javax.persistence.Id
+import javax.persistence.Table
+import javax.validation.constraints.NotNull
+
+@Entity
```

Comment:
The entity class should not be the data transfer object that is used in the rest endpoint.
Why don't you use separate these aspects into two classes and transform the dto into the entity and vice versa.